protocols/imap/src/main/java/org/apache/james/imap/processor/LoginProcessor.java

```diff
@@ -49,20 +49,24 @@ public LoginProcessor(ImapProcessor next, MailboxManager mailboxManager, StatusR
 
     @Override
     protected void processRequest(LoginRequest request, ImapSession session, Responder responder) {
-            // check if the login is allowed with LOGIN command. See IMAP-304
-            if (session.isPlainAuthDisallowed() && session.isTLSActive() == false) {
-                no(request, responder, HumanReadableText.DISABLED_LOGIN);
-            } else {
-                doAuth(noDelegation(request.getUserid(), request.getPassword()),
-                    session, request, responder, HumanReadableText.INVALID_LOGIN);
-            }
+        // check if the login is allowed with LOGIN command. See IMAP-304
+        boolean isPlainAuthDisabled = !session.isPlainAuthEnabled();
+        boolean isPlainAuthDisabledWhenNotEncrypted = session.isPlainAuthDisallowed() && !session.isTLSActive();
```

Comment:
The boolean expression looks like the same than for AuthProcessor.

Maybe we could move it in the IMAP session for reuse (eg using a default method?)